### PR TITLE
[stable/traefik] feat: add priorityClassName for pods

### PR DESCRIPTION
**Pull request by Art3mK**
Signed-off-by: Art3mK <artem@kayalaynen.ru>

#### What this PR does / why we need it:

This PR allows user to specify [PriorityClass](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/) for traefik pods

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/values.yaml [hunk 1]
```diff
@@ -16,6 +16,8 @@ podDisruptionBudget: {}
   # maxUnavailable: 1
   # minAvailable: 2
 
+priorityClassName: ""
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: Can this be commented out by default?

File: stable/traefik/templates/deployment.yaml [hunk 2]
```diff
@@ -53,6 +53,9 @@ spec:
       {{- if .Values.affinity }}
       affinity:
 {{ toYaml .Values.affinity | indent 8 }}
+      {{- end }}
+      {{- if .Values.priorityClassName }}
+      priorityClassName: "{{ .Values.priorityClassName }}"
```
Comment (dtomcej) [inline stable/traefik/templates/deployment.yaml]:
```suggestion
       priorityClassName: {{ .Values.priorityClassName | quote }}
```

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/values.yaml (hunk 1 above)
Comment (Art3mK) [reply stable/traefik/values.yaml]: all right, although it's not required, as `if` evaluates to false with empty strings

File: stable/traefik/templates/deployment.yaml (hunk 2 above)
Comment (Art3mK) [reply stable/traefik/templates/deployment.yaml]: done

**Comment by Art3mK** [comment]
ping @dtomcej, or @mmatur maybe?

**Pull request merged**